deps/build.jl

```diff
@@ -1,48 +1,80 @@
+const hsl_ma97_sha256 = "b91552164311add95f7228d1397a747611f08ffdc86a100df58ddfcedfdc7ca7"
+const hsl_ma97_archive = joinpath(@__DIR__, "downloads", "hsl_ma97-$hsl_ma97_version.tar.gz")
 
-@static if Sys.isapple()
-  so = "dylib"
-  all_load = "-all_load"
-  noall_load = "-noall_load"
-  using Homebrew
-end
+const so         = Sys.isapple() ? "dylib" : "so"
+const all_load   = Sys.isapple() ? "-all_load" : "--whole-archive"
+const noall_load = Sys.isapple() ? "-noall_load" : "--no-whole-archive"
+
+hsl_archives = [hsl_ma97_archive]
 
-hsl_modules = @compat Dict()
+if any(isfile.(hsl_archives))
+  products = Product[
+      LibraryProduct(prefix, ["libopenblas", "libopenblas64_"], :libopenblas),
+      LibraryProduct(prefix, ["libcoinmetis"], :libcoinmetis),
+  ]
 
-here = dirname(@__FILE__)
+  download_info = Dict{Symbol,Dict}()
 
-blas_and_metis_built = false
+  include("dep_openblas.jl")
+  include("dep_metis.jl")
 
-function build_blas_and_metis()
-  global blas_and_metis_built
-  if !blas_and_metis_built
-    @info "building blas"
-    include(joinpath(here, "build_blas.jl"))
-    @info "building metis"
-    include(joinpath(here, "build_metis4.jl"))
-    blas_and_metis_built = true
+  # Asserting every product has a download_info
+  for p in products
+    @assert haskey(download_info, p.variable_name)
   end
-end
 
-const hsl_ma97_version = "2.4.0"
-const hsl_ma97_sha256 = "b91552164311add95f7228d1397a747611f08ffdc86a100df58ddfcedfdc7ca7"
-const hsl_ma97_archive = joinpath(here, "downloads", "hsl_ma97-$hsl_ma97_version.tar.gz")
+  # Install unsatisfied or updated dependencies:
+  for p in products
+    vname = p.variable_name
+    unsatisfied = !satisfied(p; verbose=verbose)
+    dl_info = choose_download(download_info[vname], platform_key_abi())
+    if dl_info === nothing && unsatisfied
+      # If we don't have a compatible .tar.gz to download, complain.
+      # Alternatively, you could attempt to install from a separate provider,
+      # build from source or something even more ambitious here.
+      error("Your platform (\"$(Sys.MACHINE)\", parsed as \"$(triplet(platform_key_abi()))\") is not supported by this package because of dependency $vname")
+    end
 
-@info "looking for $hsl_ma97_archive"
-if isfile(hsl_ma97_archive)
-  @info "hsl_ma97 found"
-  build_blas_and_metis()
-  include("build_hsl_ma97.jl")
+    # If we have a download, and we are unsatisfied (or the version we're
+    # trying to install is not itself installed) then load it up!
+    if unsatisfied !! !isinstalled(dl_info...; prefix=prefix)
+      # Download and install binaries
+      install(dl_info...; prefix=prefix, force=true, verbose=verbose)
+    end
+  end
+
+  # fixup library install paths on macOS
+  usrdir = joinpath(@__DIR__, "usr")
+  libdir = joinpath(usrdir, "lib")
+  if Sys.isapple()
+    libs = filter(lib -> endswith(lib, ".dylib"), readdir(libdir))
+    for lib in libs
+      libfull = joinpath(libdir, lib)
+      run()
```

Comment:
Here is the problem. The command disappeared...

Reply:
yeap. Also, I'm getting the same Segmentation Fault problem. I just put the code missing and it is not working: That's the message
```hash
signal (11): Segmentation fault: 11
in expression starting at /Users/lrsantos/dev/JSO/HSL.jl/deps/build.jl:17
unknown function (ip: 0x11dce4bcf)
Allocations: 7076085 (Pool: 7074397; Big: 1688); GC: 15
```

Reply:
My bad. It is working now. However, one has to include the command.

Reply:
Fixed by #25